src/RNTupleWriter.cc

```diff
@@ -107,29 +113,46 @@ void RNTupleWriter::writeFrame(const podio::Frame& frame, const std::string& cat
     }
   }
 
+#if ROOT_VERSION_CODE >= ROOT_VERSION(6, 31, 0)
+  auto entry = m_categories[category].writer->GetModel().CreateBareEntry();
+#else
   auto entry = m_categories[category].writer->GetModel()->CreateBareEntry();
+#endif
```

Comment:
```suggestion
  auto entry = m_categories[category].writer->GetModel()->CreateBareEntry();
```

Reply:
The reason there is a preprocessor check here is because `GetModel` changed from returning a pointer to returning a reference. I had a version using `std::invoke` in there that works for all versions, but for a future removal this is easier to manage, I think.

Reply:
Ah yes, missed the . is a ->